packages/react/toast/src/Toast.tsx

```diff
@@ -173,36 +173,70 @@ const ToastViewport = React.forwardRef<ToastViewportElement, ToastViewportProps>
     React.useEffect(() => {
       const wrapper = wrapperRef.current;
       const viewport = ref.current;
-      if (wrapper && viewport) {
+      if (hasToasts && wrapper && viewport) {
         const handlePause = () => {
-          const pauseEvent = new CustomEvent(VIEWPORT_PAUSE);
-          viewport.dispatchEvent(pauseEvent);
-          context.isClosePausedRef.current = true;
+          if (!context.isClosePausedRef.current) {
+            const pauseEvent = new CustomEvent(VIEWPORT_PAUSE);
+            viewport.dispatchEvent(pauseEvent);
+            context.isClosePausedRef.current = true;
+          }
         };
 
         const handleResume = () => {
-          const resumeEvent = new CustomEvent(VIEWPORT_RESUME);
-          viewport.dispatchEvent(resumeEvent);
-          context.isClosePausedRef.current = false;
+          if (context.isClosePausedRef.current) {
+            const resumeEvent = new CustomEvent(VIEWPORT_RESUME);
+            viewport.dispatchEvent(resumeEvent);
+            context.isClosePausedRef.current = false;
+          }
+        };
+
+        const isFocusProxy = (element: HTMLElement | null) => {
+          const tailProxyNode = tailFocusProxyRef.current;
+          const headProxyNode = headFocusProxyRef.current;
+          return element === tailProxyNode || element === headProxyNode;
+        };
+
+        const handleFocusOutResume = (event: FocusEvent) => {
```

Comment:
When focusing away from `window` we were seeing duplicate events as there was an overlap between `focusout` and `window.blur`, they both fire resulting in `onPause` -> `onResume` -> `onPause`.

As a consumer I wouldn't expect to see this, if they are already paused then the handlers shouldn't fire until that perceivably changes.